daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DPath.scala

```diff
@@ -195,26 +203,26 @@ final class RuntimeExpressionDPath[T <: AnyRef](qn: NamedQName, tt: NodeInfo.Kin
     recipe.run(dstate)
   }
 
-  private def processForwardExpressionResults(dstate: DState): AnyRef = {
-    val v: AnyRef = {
+  private def processForwardExpressionResults(dstate: DState): Maybe[T] = {
```

Comment:
Not sure why this changed. Looking at the code again, just AnyRef should still be fine